bukkit-helper-121-3/src/main/java/org/dynmap/bukkit/helper/v121_3/MapChunkCache121_3.java

```diff
@@ -31,6 +36,22 @@ public MapChunkCache121_3(GenericChunkCache cc) {
         super(cc);
     }
 
+    @Override
+    protected Supplier<GenericChunk> getLoadedChunkAsync(DynmapChunk chunk) {
+        CompletableFuture<SerializableChunkData> chunkData = CompletableFuture.supplyAsync(() -> {
+            CraftWorld cw = (CraftWorld) w;
+            Chunk c = cw.getHandle().getChunkIfLoaded(chunk.x, chunk.z);
+            if (c == null || !c.q) { //!c.loaded
+                return null;
+            }
+            return SerializableChunkData.a(cw.getHandle(), c); //SerializableChunkData.copyOf
+        }, ((CraftServer) Bukkit.getServer()).getServer());
+        return () -> {
+            NBTTagCompound nbt = chunkData.join().a(); // SerializableChunkData.write
```

Comment:
The CompletableFuture might return null, so this throws a NullPointerException if chunkData.join() is null. Assign that to a temporary var and return null if the SerializableChunkData, and that will fix it.